source/non_matching/coupling.cc

```diff
@@ -40,6 +40,92 @@
 DEAL_II_NAMESPACE_OPEN
 namespace NonMatching
 {
+  namespace internal
+  {
+    /**
+     * Given two triangulations, the first immersed inside the other, this
+     * function computes and returns the real-space quadrature points of the immersed
+     * triangulation.
```

Comment:
Actually, it's still not clear to me what the function returns. You say "the real-space quadrature points", which I take as "the locations of quadrature points of the immersed triangulation in real space". If that's what you do, then you wouldn't need the imbedding triangulation because it doesn't factor here, so I probably don't quite understand.

But also, the function doesn't just return a list of points, but in fact a `std::pair<std::vector<Point<spacedim>>, std::vector<unsigned int>>`. Can you explain what the two parts mean?